src/annotator/config/index.js

```diff
@@ -53,67 +61,104 @@ function configurationKeys(appContext) {
       return contexts.notebook;
     case 'all':
       // Complete list of configuration keys used for testing.
-      return [...contexts.annotator, ...contexts.sidebar, ...contexts.notebook];
+      return [
+        ...contexts.annotator,
+        ...contexts.sidebar,
+        ...contexts.notebook,
+        // "experimental" currently has no uses in any app contexts, but is included
+        // for test coverage
+        'experimental',
+      ];
     default:
       throw new Error(`Invalid application context used: "${appContext}"`);
   }
 }
 
 /**
- * Reads the Hypothesis configuration from the environment.
- *
- * @param {string[]} settingsKeys - List of settings that should be returned.
- * @param {Window} window_ - The Window object to read config from.
+ * The definition for configuration sources, their default values
+ * @return {ConfigDefinitionMap}
  */
-function configFrom(settingsKeys, window_) {
-  const settings = settingsFrom(window_);
-  const allConfigSettings = {
-    annotations: settings.annotations,
-    appType: settings.hostPageSetting('appType', {
-      allowInBrowserExt: true,
-    }),
-    branding: settings.hostPageSetting('branding'),
+function configDefinitions(settings) {
```

Comment:
Logically `configDefinitions` returns a static description of all supported configuration properties. Currently this is wrapped inside a function so that you can pass in `settings`. If however the `valueFn` took `settings` as an argument then you could make `configDefinitions` a static variable:

```js
const configDefinitions = {
  annotations: {
    valueFn: settings => settings.annotations,
  },
  ...
}
```

The advantage of this would be primarily to make it obvious that the shape of `configDefinitions` is static and only the part involving reading the value is dynamic. A side benefit is that you could potentially reference the shape of `configDefinitions` in types more easily.

Reply:
I think that's fine and it makes total sense. In the next set of commits in the third PR, I had been passing the key name in as well to reduce some redundancy, but maybe there is a way we can still accomplish that and still pass settings too.